src/test/java/de/muenchen/isi/domain/service/calculation/SobonursaechlicheWohneinheitenServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package de.muenchen.isi.domain.service.calculation;
+
+import static de.muenchen.isi.TestConstants.EOF;
+import static de.muenchen.isi.TestConstants.FF;
+import static de.muenchen.isi.TestConstants.MM;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+import de.muenchen.isi.domain.model.BaurateModel;
+import de.muenchen.isi.domain.model.FoerderartModel;
+import de.muenchen.isi.domain.model.FoerdermixModel;
+import de.muenchen.isi.domain.model.calculation.WohneinheitenProFoerderartProJahrModel;
+import de.muenchen.isi.infrastructure.entity.enums.lookup.SobonOrientierungswertJahr;
+import de.muenchen.isi.infrastructure.repository.stammdaten.StaedtebaulicheOrientierungswertRepository;
+import de.muenchen.isi.infrastructure.repository.stammdaten.UmlegungFoerderartenRepository;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class SobonursaechlicheWohneinheitenServiceTest {
+
+    private SobonursaechlicheWohneinheitenService sobonursaechlicheWohneinheitenService;
+
+    @Mock
+    private UmlegungFoerderartenRepository umlegungFoerderartenRepository;
+
+    @Mock
+    private StaedtebaulicheOrientierungswertRepository staedtebaulicheOrientierungswertRepository;
+
+    @BeforeEach
+    public void beforeEach() {
+        final var foerdermixUmlageService = new FoerdermixUmlageService(umlegungFoerderartenRepository);
+        this.sobonursaechlicheWohneinheitenService =
+            new SobonursaechlicheWohneinheitenService(
+                foerdermixUmlageService,
+                staedtebaulicheOrientierungswertRepository
+            );
+        Mockito.reset(umlegungFoerderartenRepository, staedtebaulicheOrientierungswertRepository);
+    }
+
+    @Test
+    void calculateSobonursaechlicheWohneinheitenTest() {
+        FoerdermixUmlageServiceTest.fillUmlegungFoerderartenRepository(umlegungFoerderartenRepository);
+        PlanungsursaechlicheWohneinheitenServiceTest.fillStaedtebaulicheOrientierungswertRepository(
+            staedtebaulicheOrientierungswertRepository
+        );
+
+        // Fördermixe
+        final var FF50 = new FoerderartModel();
+        FF50.setBezeichnung(FF);
+        FF50.setAnteilProzent(new BigDecimal("0.5"));
```

Comment:
Die Tatsächlichen Förderarten sind als ganzzahligeProzentwerte (`0,5555` -> `55,55`) in der Datenbank persistiert.

ggf. somit die Berechnungslogik anpassen.